pennylane/math/quantum.py

```diff
@@ -1033,6 +1033,12 @@ def _denman_beavers_iterations(mat, max_iter=100, tol=1e-13):
         LinAlgError: If matrix inversion fails
         ValueError: If NaN values or overflow are encountered during computation
     """
+    if mat.shape == (1, 1):
+        value = mat[0, 0]
+        returned_value = np.sqrt(value)
+        returned_mat = mat.copy()
+        returned_mat[0, 0] = returned_value
+        return returned_mat

```

Comment:
`return np.sqrt(mat)` should work here, right?

Reply:
No, mat is still a 1by1 matrix wrapped by `scipy.sparse` so `np.sqrt(mat)` will fail finding a function.
Maybe let me try register yet another customized `sqrt` here for `scipy` interface

Reply:
Oh no I saw in scipy src they've dispatched lots of unary np funcs for direct elementwise data manipulation: https://github.com/scipy/scipy/blob/5cf433589639f28cbf5c3911061b07f0c6d6bd68/scipy/sparse/_data.py#L139
https://github.com/scipy/scipy/blob/5cf433589639f28cbf5c3911061b07f0c6d6bd68/scipy/sparse/_base.py#L57-L60

I guess our `from autoray import numpy as np` broke this part of dispatch that's why I could never pass with `np.sqrt` here. I'll try direct import sqrt from native numpy

Reply:
https://github.com/PennyLaneAI/pennylane/pull/6963/commits/aff3a8d7fd1364d826e17fd623bdf89ca89f479c